Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3439,18 +3439,31 @@ ErrorOr<RefPtr<StyleValue>> Parser::parse_dynamic_value(ComponentValue const& co
     if (component_value.is_function()) {
         auto const& function = component_value.function();
 
-        if (function.name().equals_ignoring_ascii_case("calc"sv))
-            return parse_calculated_value(function.values());
-
         if (function.name().equals_ignoring_ascii_case("var"sv)) {
             // Declarations using `var()` should already be parsed as an UnresolvedStyleValue before this point.
             VERIFY_NOT_REACHED();
         }
+
+        auto function_node = TRY(parse_a_function_node(function));
+        if (!function_node)
+            return nullptr;
+
+        auto function_type = function_node->resolved_type().value();
+        return CalculatedStyleValue::create(function_node.release_nonnull(), function_type);
     }
 
     return nullptr;
 }
 
+ErrorOr<OwnPtr<CalculationNode>> Parser::parse_a_function_node(Function const& function)
```

Comment:
Minor name nit-pick: Not all functions are calculation-related, could we mention in the name that it's a math/calculation-function? I might be overthinking this.